dags/atd_moped_data_tracker_sync.py

```diff
@@ -0,0 +1,72 @@
+# Test locally with: docker compose run --rm airflow-cli dags test atd_moped_data_tracker_sync
+
+import os
+
+from airflow.models import DAG
+from airflow.operators.docker_operator import DockerOperator
+from pendulum import datetime, duration
+
+from utils.onepassword import get_env_vars_task
+from utils.slack_operator import task_fail_slack_alert
+from utils.knack import get_date_filter_arg
+
+DEPLOYMENT_ENVIRONMENT = os.getenv("ENVIRONMENT", "development")
+
+DEFAULT_ARGS = {
+    "owner": "airflow",
+    "depends_on_past": False,
+    "start_date": datetime(2015, 1, 1, tz="America/Chicago"),
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 0,
+    "retry_delay": duration(minutes=5),
+    "on_failure_callback": task_fail_slack_alert,
+}
+
+REQUIRED_SECRETS = {
+    "HASURA_ENDPOINT": {
+        "opitem": "Moped Hasura Admin",
+        "opfield": f"{DEPLOYMENT_ENVIRONMENT}.Endpoint",
+    },
+    "HASURA_ADMIN_SECRET": {
+        "opitem": "Moped Hasura Admin",
+        "opfield": f"{DEPLOYMENT_ENVIRONMENT}.Admin Secret",
+    },
+    "KNACK_DATA_TRACKER_APP_ID": {
+        "opitem": "Knack AMD Data Tracker",
+        "opfield": f"{DEPLOYMENT_ENVIRONMENT}.appId",
+    },
+    "KNACK_DATA_TRACKER_API_KEY": {
+        "opitem": "Knack AMD Data Tracker",
+        "opfield": f"{DEPLOYMENT_ENVIRONMENT}.apiKey",
+    },
+}
+
+
+with DAG(
+    dag_id="atd_moped_data_tracker_sync",
+    description="sync Moped project data to Knack Data Tracker projects table",
+    default_args=DEFAULT_ARGS,
+    schedule_interval="0 * * * *" if DEPLOYMENT_ENVIRONMENT == "production" else None,
+    dagrun_timeout=duration(minutes=30),
+    tags=["repo:atd-moped", "moped", "data-tracker", "knack"],
+    catchup=False,
+) as dag:
+    docker_image = "atddocker/atd-moped-etl-data-tracker-sync:production"
+
+    date_filter_arg = get_date_filter_arg()
+
+    env_vars = get_env_vars_task(REQUIRED_SECRETS)
+
+    t1 = DockerOperator(
+        task_id="data_tracker_sync",
+        image=docker_image,
+        auto_remove=True,
+        command=f"python data_tracker_sync.py {date_filter_arg}",
```

Comment:
Looking at the code this invokes, it looks like this may need a `-d` or `--date` to comply to the format argparse wants. Here's the code in moped: https://github.com/cityofaustin/atd-moped/blob/main/moped-etl/data-tracker-sync/data_tracker_sync.py#L232-L238.

Reply:
This has been really helpful to think through, and I think the records fall into three categories (so far):
- Moped projects not synced to Knack (covered by script)
- Moped projects synced to Knack and updated since the last successful run (covered)
- Moped project already synced to Knack and updated before this script existed and before the look-back time of the very first run (not covered)

So, I agree that we need to run this once with a look-back time far back enough to cover that third category. Something like what [this helper](https://github.com/cityofaustin/atd-airflow/blob/production/dags/utils/time.py) does.

Reply:
jinx @johnclary 😄

Reply:
thanks Mike!!